app/grandchallenge/groups/views.py

```diff
@@ -61,9 +68,50 @@ def get_queryset(self):
             get_user_model()
             .objects.order_by("username")
             .exclude(username=settings.ANONYMOUS_USER_NAME)
+            .annotate(
+                full_name=Concat(
+                    "first_name",
+                    Value(" "),
+                    "last_name",
+                    output_field=CharField(),
+                )
+            )
+            .select_related("verification", "user_profile")
         )
 
         if self.q:
-            qs = qs.filter(username__istartswith=self.q)
+            qs = qs.filter(
+                Q(username__icontains=self.q)
+                | Q(email__icontains=self.q)
+                | Q(full_name__icontains=self.q)
+                | Q(verification__email=self.q)
+            )
 
         return qs
+
+    def get_result_label(self, result):
+
+        try:
+            is_verified = result.verification.is_verified
+        except Verification.DoesNotExist:
+            is_verified = False
+
+        if is_verified:
+            return format_html(
+                '<img src="{}" width ="20" height ="20" style="vertical-align:top"> '
+                "&nbsp; <b>{}</b> &nbsp; {} &nbsp;"
+                '<i class="fas fa-user-check text-success" '
+                'title="Verified email address at {}">',
+                mark_safe(result.user_profile.get_mugshot_url()),
```

Comment:
Do you really need `mark_safe` here? I think that it is not needed eg, https://github.com/comic/grand-challenge.org/blob/master/app/grandchallenge/profiles/templatetags/profiles.py#L24

Reply:
It's not strictly necessary because the mugshot urls on production do not contain html unsafe characters. But we noticed during testing that the mugshot url that it uses then contains a `&` which will be converted to `&amp;` by `format_html` which would make the url invalid. So I think it should also be added on the line you refer to. 
Although we should be careful with this because it could potentially introduce an XSS vulnerability (I guess the mugshot url is user input in essence?), so maybe since it is not necessary for production we should remove it to be safe.

Reply:
I'm pretty sure that the user file names are cleaned and normalized so that they do not contain special symbols, and yes, better to remove it than risk an XSS vulnerability.

Reply:
ok, removing it sounds good, but then we again have the issue that the test no longer works, so is it ok then to not test whether the mugshot url is presented?

Reply:
Yes, use `html_escape` in the tests. This is automatically applied in Django templates, but sometimes we need to be explicit about it.